dirsrvtests/tests/suites/logging/logconv_test.py

```diff
@@ -0,0 +1,980 @@
+
+    test_env.truncate_logs()
+    test_env.run_operation(test_env.operation)
+
+def _test_persistant_search(test_env):
+
+    def _persistant_search(test_env):
+        psc = PersistentSearchControl()
+
+        msgid = test_env.inst.search_ext(
+            base=DEFAULT_SUFFIX,
+            scope=ldap.SCOPE_ONELEVEL,
+            filterstr='(objectClass=person)',
+            attrlist=['uid'],
+            serverctrls=[psc]
+        )
+
+        test_env.inst.abandon(msgid)
+
+    test_env.operation = _persistant_search
+
+    test_env.expected = {
+        "searches": 1,
+        "persistent_searches": 1,
+        "abandoned": 1,
+        "operations": 2,
+        "results": 1, # abandoned op counted as a result
+    }
+
+    test_env.truncate_logs()
+    test_env.run_operation(test_env.operation)
+
+
+def _test_simple_bind(test_env):
+
+    def _simple_bind(test_env):
+
+
+        users = UserAccounts(test_env.inst, DEFAULT_SUFFIX)
+        user = users.create_test_user(uid=1, gid=1)
+        user.set("userPassword", "password")
+
+        conn = ldap.initialize("ldap://%s:%s" % (HOST_STANDALONE, PORT_STANDALONE))
+        conn.simple_bind_s(user.dn, "password")
+        conn.unbind_s()
+        user.delete()
+
+    test_env.operation = _simple_bind
+
+    test_env.expected = {
+        "adds": 1,
+        "mods": 1,
+        "deletes": 1,
+        "binds": 1,
+        "unbinds": 1,
+        "searches": 2,
+        "operations": 6,
+        "results": 6,
+        "total_connections": 1,
+        "concurrent_conns": 1,
+        "ldap_conns": 1,
+        "fds_taken": 1,
+        "fds_returned": 1
+    }
+
+    test_env.truncate_logs()
+    test_env.run_operation(test_env.operation)
+
+def _test_anon_bind(test_env):
+
+    def _anon_bind(test_env):
+        conn = Anonymous(test_env.inst).bind()
+        conn.unbind()
+
+    test_env.operation = _anon_bind
+
+    test_env.expected = {
+        "binds": 1,
+        "unbinds": 1,
+        "anon_binds": 1,
+        "searches": 1,
+        "operations": 2,
+        "results": 2,
+        "ldap_conns": 1,
+        "concurrent_conns": 1,
+        "total_connections": 1,
+        "fds_taken": 1,
+        "fds_returned": 1
+    }
+
+    test_env.truncate_logs()
+    test_env.run_operation(test_env.operation)
+
+def _test_ldapi(test_env):
+
+    def _ldapi_search(tetest_envt):
+        test_env.inst.config.set('nsslapd-ldapilisten', 'on')
+        test_env.inst.config.set('nsslapd-ldapiautobind', 'on')
+        test_env.inst.config.set('nsslapd-ldapifilepath', f'/var/run/slapd-{test_env.inst.serverid}.socket')
+        test_env.inst.restart()
+        ldapi_socket = test_env.inst.config.get_attr_val_utf8('nsslapd-ldapifilepath').replace('/', '%2F')
+        cmd = [
+            "ldapsearch",
+            "-H", f"ldapi://{ldapi_socket}",
+            "-Y", "EXTERNAL",
+            "-b", "dc=example,dc=com"
+        ]
+        subprocess.run(cmd, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
```

Comment:
Not capturing output makes debugging failures difficult (in case, something will happen during the search - we can see the issue right away).

Reply:
> Not capturing output makes debugging failures difficult (in case, something will happen during the search - we can see the issue right away).

Good point